randovania/layout/preset_migration.py

```diff
@@ -1163,6 +1163,16 @@ def fix(identifier: dict) -> None:
             fix(it)
 
 
+def _migrate_v103(preset: dict, game: RandovaniaGame) -> None:
+    if game != RandovaniaGame.METROID_PRIME:
+        return
+
+    dlr = preset["configuration"]["dock_rando"]["mode"] != "vanilla"
+    legacy = preset["configuration"]["legacy_mode"]
+    preset["configuration"]["blast_shield_lockon"] = dlr and not legacy
+    preset["configuration"]["blue_save_doors"] = dlr and not legacy
```

Comment:
With this, a preset that has DLR disabled but blue save doors on will be migrated to disabled. Is this intended?